src/js/components/Text/Text.js

```diff
@@ -73,9 +73,11 @@ const Text = forwardRef(
         {...rest}
         ref={textRef}
       >
-        <TextContext.Provider value={textContextValue}>
-          {children}
-        </TextContext.Provider>
+        {children !== undefined ? (
+          <TextContext.Provider value={textContextValue}>
+            {children}
+          </TextContext.Provider>
+        ) : undefined}
```

Comment:
Slight simplification
```suggestion
        {children && (
          <TextContext.Provider value={textContextValue}>
            {children}
          </TextContext.Provider>
        )}
```

Reply:
same as  https://github.com/grommet/grommet/pull/6676#discussion_r1131370082